### ISSUE TITLE
Question about discriminator loss

### ISSUE THREAD

**Issue by Khanhnn00** [Lornatang/SRGAN-PyTorch#12]
Hi there, thank you for your dedicated work
However, I found that while training srgan, in discriminator, you said: # The real sample label is 1, and the generated sample label is 0. Then you used d_loss = (d_loss_real + d_loss_fake) / 2. Can you please kindly explain me this loss and the motivation behind setting real sample and generated sample? I may be blind but tbh I did not see it in the paper.

Thank you, best regards

**Comment by Lornatang** [comment]
Actually `d_loss = d_loss_real + d_loss_fake`,  But I set it up to prevent the discriminator from training too well.

**Comment by Khanhnn00** [comment]
thanks for your response. but I am still curious about the real sample and generated sample, because I didnot see it anywhere in paper. can u explain me it?

**Comment by Lornatang** [comment]
Where are you curious about where to generate samples or real samples

**Comment by Khanhnn00** [comment]
Hi, sorry for the uninformative question.
I see that you used: d_loss_real = adversarial_criterion(real_output, real_label), with real_label is marked as 1 and fake_label is marked as 0 in contrary. I don't see it in any equations available in the paper, so I'm a little bit confused. Does it equal to the equation (2) in the paper of SRGAN?
Thanks for your patience and warm help. I really appreciate

**Comment by Lornatang** [comment]
See Goodfellow GAN paper.
